selftests/functional/test_basic.py

```diff
@@ -50,6 +50,17 @@ def test(self):
         self.log.info(hello())
 '''
 
+INVALID_PYTHON_TEST = '''
+from avocado import Test
+
+class MyTest(Test):
+
+    non_existing_variable_causing_crash
+
+    def test(self):
```

Comment:
That's what I thought, but as the traceback contains the original information, it actually reports the original test method.

Try it on `multiple_tests` test. I modified the `test_hello` to crash and the result is:

```
START examples/tests/multiple_tests.py:MultipleTests.test_hello

Reproduced traceback from: /home/medic/Work/Projekty/avocado/avocado/avocado/core/test.py:389
Traceback (most recent call last):
  File "examples/tests/multiple_tests.py", line 18, in test_hello
    asdf
NameError: global name 'asdf' is not defined

Traceback (most recent call last):

  File "/home/medic/Work/Projekty/avocado/avocado/avocado/core/test.py", line 444, in _run_avocado
    raise test_exception

NameError: global name 'asdf' is not defined

ERROR examples/tests/multiple_tests.py:MultipleTests.test_hello -> NameError: global name 'asdf' is not defined
```

So the `TestError`'s method name does not matter at all and the unittest should check if the test name is correct (`examples/tests/multiple_tests.py:MultipleTests.test_hello` in this example).